File: Leetcode/Random Questions/Easy/DefangingAnIPAddress/DefangingAnIPAddress.cpp
#include<iostream>
using namespace std;
#include<string>
class Solution {
public:
    string defangIPaddr(string address) {
        string solution;
        for(char &itr: address)
        {
            if(itr == '.')
                solution += "[.]";
            else
                solution += itr;
        }
        return solution;          
    }
};

// Space complexity - O(n) --> Storing in a string
// Time Complexity - O(n)

File: LeetCode/Algorithms/Easy/DefangingAnIPAddress/DefangingAnIPAddress.py
class Solution:
    def defangIPaddr(self, address: str) -> str:
        a = ""
        for i in range(len(address)):
            if address[i] == '.':
                a += '[.]'
            elif address[i] != '.':
                a += address[i]
        return a

# Time Complexity - O(n)
# Space Complexity - O(n)